x-pack/plugins/security_solution/public/risk_score/components/risk_details_tab_body/index.tsx

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiButton, EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import React, { useCallback, useMemo } from 'react';
+import styled from 'styled-components';
+import { RISKY_HOSTS_DASHBOARD_TITLE, RISKY_USERS_DASHBOARD_TITLE } from '../../constants';
+import { EnableRiskScore } from '../enable_risk_score';
+import { useDeepEqualSelector } from '../../../common/hooks/use_selector';
+import type { State } from '../../../common/store';
+import { hostsModel, hostsSelectors } from '../../../hosts/store';
+import { usersSelectors } from '../../../users/store';
+import { RiskInformationButtonEmpty } from '../risk_information';
+import * as i18n from './translations';
+
+import { useQueryInspector } from '../../../common/components/page/manage_query';
+import { RiskScoreOverTime } from '../risk_score_over_time';
+import { TopRiskScoreContributors } from '../../../common/components/top_risk_score_contributors';
+import { useQueryToggle } from '../../../common/containers/query_toggle';
+import {
+  HostRiskScoreQueryId,
+  useHostRiskScore,
+  UserRiskScoreQueryId,
+  useUserRiskScore,
+} from '../../containers';
+import type { HostRiskScore, UserRiskScore } from '../../../../common/search_strategy';
+import { buildEntityNameFilter, RiskScoreEntity } from '../../../../common/search_strategy';
+import type { UsersComponentsQueryProps } from '../../../users/pages/navigation/types';
+import type { HostsComponentsQueryProps } from '../../../hosts/pages/navigation/types';
+import { useDashboardButtonHref } from '../../../common/hooks/use_dashboard_button_href';
+import { RiskScoresNoDataDetected } from '../risk_score_onboarding/risk_score_no_data_detected';
+
+const StyledEuiFlexGroup = styled(EuiFlexGroup)`
+  margin-top: ${({ theme }) => theme.eui.euiSizeL};
+`;
+
+type ComponentsQueryProps = HostsComponentsQueryProps | UsersComponentsQueryProps;
+
+const getDashboardTitle = (riskEntity: RiskScoreEntity) =>
+  riskEntity === RiskScoreEntity.host ? RISKY_HOSTS_DASHBOARD_TITLE : RISKY_USERS_DASHBOARD_TITLE;
+
+const RiskDetailsTabBodyComponent: React.FC<
+  Pick<ComponentsQueryProps, 'startDate' | 'endDate' | 'setQuery' | 'deleteQuery'> & {
+    entityName: string;
+    riskEntity: RiskScoreEntity;
+  }
+> = ({ entityName, startDate, endDate, setQuery, deleteQuery, riskEntity }) => {
+  const entity = useMemo(
+    () =>
+      riskEntity === RiskScoreEntity.host
```

Comment:
Here, it is desfined a hook conditionally, which breaks [react hooks rules](https://reactjs.org/docs/hooks-rules.html). So if the `riskEntity` property changes, the component won't work. 

You could export `useRiskScore` and call it instead of calling `useHostRiskScore` and `useUserRiskScore` conditionalt.